core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -680,12 +680,15 @@ abstract class AbstractFetcherThread(name: String,
        */
       val offsetAndEpoch = leader.fetchEarliestOffset(topicPartition, currentLeaderEpoch)
       val leaderStartOffset = offsetAndEpoch.offset
-      warn(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to current " +
-        s"leader's start offset $leaderStartOffset")
       val offsetToFetch = Math.max(leaderStartOffset, replicaEndOffset)
+      warn(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to " +
```

Comment:
> Why do we have to log this at WARN level? Can we change it to the INFO level?

Thanks for your review! Since the fetchOffsetAndTruncate method deals with the situation that a follower encounters an out of range during fetch, whether it is due to dirty elections that cause the log end offset of the old leader to be larger than the log end offset of the new leader requires truncate, or because the follower stopped for a long time causes the log end offset to be smaller than the leader's log start offset requires truncate, both of which belong to a type of "abnormal" processing case, so the logs in this method are all at the warn level. This is my understanding, please correct me if I am wrong, thank you! @kamalcph

Reply:
We have one WARN log when the leaderStartOffset is greater than the replicaEndOffset to let know the user that the replica log was fully truncated.

In the other case, the follower/replica won't do any truncation and it's going to fetch from the `replicaEndOffset`. With unclean leader election enabled, we cannot know whether the log has diverged b/w leader and replica at this place. I was wondering, if can we reduce the logger level to INFO since the user won't get enough information from this log.

Reply:
Thank you for your explanation. I very much agree with what you said that it is impossible to judge the possibility of the case of unclean leader election. I have changed this line of log to info. @kamalcph